gocd/bci_repo_publish.py

```diff
@@ -131,35 +134,28 @@ def run(self, version, token=None):
             return
 
         # Check openQA results
-        openqa_passed = True
         for pkg in packages:
-            passed = 0
-            pending = 0
-            failed = 0
             for job in self.openqa_jobs_for_product(arch=pkg['arch'], version=version,
                                                     build=pkg['built_version']):
-                if job['result'] in ('passed', 'softfailed'):
-                    passed += 1
-                elif job['result'] == 'none':
+                if job['result'] == 'none':
                     self.logger.info(f'https://openqa.suse.de/tests/{job["id"]} pending')
-                    pending += 1
-                else:
+                    break
+                if job['result'] not in ('passed', 'softfailed'):
                     self.logger.warning(f'https://openqa.suse.de/tests/{job["id"]} failed')
-                    failed += 1
+                    break
+            else:
+                openqa_passed_packages.append(pkg)
```

Comment:
Please add a comment here how the `for - else` construct works. It is so seldomly used, that many have no idea what it does.

Reply:
+1. I had to double-check what the behaviour of `for - else` is.

Due to https://github.com/openSUSE/openSUSE-release-tools/pull/3029#discussion_r1374251819, the `break` should be removed which means the `for - else` way would no longer work anyway.